### PR TITLE
refactor(rust): use temporary secure channels on cloud and enroll api endpoints

### PR DESCRIPTION
**Pull request by adrianbenavides**
Tested with basic workflow:
- create node
- create identity
- enroll --auth0
- space create
- project create

### CODE REVIEW

File: implementations/rust/ockam/ockam_api/src/nodes/service.rs [hunk 1]
```diff
@@ -104,14 +107,41 @@ impl NodeMan {
...
 
 impl NodeMan {
-    pub(crate) fn api_service_route(&self, route: &str, api_service: &str) -> Result<Route> {
-        let mut route = Route::parse(route)
-            .ok_or_else(|| ApiError::generic(&format!("Invalid route: {route}")))?;
-        let route: Route = route.modify().append(api_service).into();
-        Ok(route)
+    pub(crate) async fn secure_channel(&self, route: impl Into<Route>) -> Result<Address> {
+        let route = route.into();
+        println!("ddd route {}", route);
```
Comment (polvorin) [inline implementations/rust/ockam/ockam_api/src/nodes/service.rs:121]: leftover

**Review by polvorin — APPROVED** [review]
looks good,  only a small leftover I pointed

File: implementations/rust/ockam/ockam_api/src/cloud/enroll.rs [hunk 2]
```diff
@@ -72,24 +72,29 @@ mod node {
...
-            let cloud_address = req_wrapper.route;
+            let cloud_route = req_wrapper.route()?;
             let req_body: Attributes = req_wrapper.req;
             let req_body = RequestEnrollmentToken::new(req_body);
 
             let label = "enrollment_token_generator";
             trace!(target: TARGET, "generating tokens");
 
-            let route = self.api_service_route(&cloud_address, "enrollment_token_authenticator")?;
+            let sc = self.secure_channel(cloud_route).await?;
```
Comment (mrinalwadhwa) [inline implementations/rust/ockam/ockam_api/src/cloud/enroll.rs:82]: note that this should use an identifier trust policy when we have a pinned identifier for the controller

**Review by mrinalwadhwa — APPROVED** [review]

File: implementations/rust/ockam/ockam_api/src/nodes/service.rs (hunk 1 above)
Comment (adrianbenavides) [reply implementations/rust/ockam/ockam_api/src/nodes/service.rs:121]: ooops thanks!

**Pull request merged**